src/controllers/authenticationController.ts

```diff
@@ -151,7 +175,31 @@ export default class AuthenticationController {
           type: "INFO",
         },
       });
-      return ResponsePermanentRedirect(redirectionUrl);
+
+      return pipe(
+        isUserElegibleForIoLoginUrlScheme(spidUser.fiscalNumber),
+        B.fold(
+          () => redirectionUrl,
+          () =>
+            pipe(
+              record
+                .collect(S.Ord)((key, value) => `${key}=${value}`)({
+                  errorCode: AGE_LIMIT_ERROR_CODE,
+                })
+                .join("&"),
+              (errorParams) =>
+                CLIENT_ERROR_REDIRECTION_URL.concat(`?${errorParams}`),
+              S.replace(defaultUrlSchemeRegex, "iologin:"),
+              UrlFromString.decode,
+              E.getOrElseW((err) => {
+                throw new Error(
+                  `Invalid url | ${readableReportSimplified(err)}`
+                );
+              })
+            )
```

Comment:
Can we reuse `redirectionUrl` in this branch too? Replacing the `defaultUrlSchemeRegex` and decoding it again should be enough.

Reply:
refactored in https://github.com/pagopa/io-backend/pull/1012/commits/d28fdcf11cd794984c55d974a73f22bb27ffb71b